File: MakiMora.Core/Entities/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MakiMora.Core.Entities
{
    public class Product : BaseEntity
    {
        [Key]
        public new Guid Id { get; set; } = Guid.NewGuid();
        
        [Required]
        [MaxLength(10)]
        public string Name { get; set; } = string.Empty;
        
        public string? Description { get; set; }
        
        [Required]
        [Column(TypeName = "decimal(10,2)")]
        public decimal Price { get; set; }
        
        [Required]
        public Guid CategoryId { get; set; }
        
        [Required]
        public Guid LocationId { get; set; }
        
        public string? ImageUrl { get; set; }
        public int? PreparationTime { get; set; } // Время приготовления в минутах
        
        public bool IsAvailable { get; set; } = true;
        public bool IsOnStopList { get; set; } = false;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        
        // Navigation properties
        [ForeignKey("CategoryId")]
        public virtual Category Category { get; set; } = null!;
        
        [ForeignKey("LocationId")]
        public virtual Location Location { get; set; } = null!;
        
        public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    }
}

File: MakiMora.Frontend/src/types/product.ts
// No need to import Category and Location since they are not used in this file

export interface Product {
  id: string;
  name: string;
  description?: string;
  price: number;
  categoryId: string;
  locationId: string;
  imageUrl?: string;
  preparationTime?: number;
  isAvailable: boolean;
  isOnStopList: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateProductRequest {
  name: string;
  description?: string;
  price: number;
  categoryId: string;
  locationId: string;
  imageUrl?: string;
  preparationTime?: number;
  isAvailable: boolean;
}

export interface UpdateProductRequest {
  name: string;
  description?: string;
  price: number;
  categoryId: string;
  imageUrl?: string;
  preparationTime?: number;
  isAvailable: boolean;
}

export interface UpdateProductAvailabilityRequest {
  isAvailable: boolean;
}

export interface UpdateProductStopListStatusRequest {
  isOnStopList: boolean;
}